### ISSUE TITLE
boolean field option is true if option is present but boolean method parameter option is false if option is present

### ISSUE THREAD

**Issue by Onedy** [remkop/picocli#1772]
Minimal example:
```
@CommandLine.Command(name = "test")
public class TestCommand implements Command {
    @Option(names = "-r")
    public boolean option1;

    @CommandLine.Command(name = "subcommand")
    void start(@Option(names = "-s") boolean option2) {
        System.out.println("-r is "+option1 +" and -s is "+option2);
    }
}
```
Command: `test -r subcommand -s`
Output: `-r is true and -s is false`

**Comment by Onedy** [comment]
It seems also that no matter what value I specify for -s option (after adding arity="1"), the value is always false.
```
@CommandLine.Command(name = "test")
public class TestCommand implements Command {
    @Option(names = "-r")
    public boolean option1;

    @CommandLine.Command(name = "subcommand")
    void start(@Option(names = "-s", arity = "1") boolean option2) {
        System.out.println("-r is "+option1 +" and -s is "+option2);
    }
}
```
Command: `test -r subcommand -s true`
Output: -r is true and -s is false

**Comment by remkop** [comment]
I tested both variants you mentioned but I am unable to reproduce the issue.

Can you try with `-Dpicocli.trace=DEBUG` and paste the output here?

**Comment by Onedy** [comment]
I don't have anymore the previous minimal example but I've created a minimal project where you can reproduce the issue with a test:
[PicocliTest.zip](https://github.com/remkop/picocli/files/9307165/PicocliTest.zip)
Just unzip and run with `./gradlew clean assemble :test --tests "test.MyTest" --info`
And look at:
```
MyTest > testIssue1772() FAILED
    org.opentest4j.AssertionFailedError: 
    expected: "------> -r is true and -s is true <------"
     but was: "------> -r is true and -s is false <------"
```

**Comment by remkop** [comment]
@Onedy Thank you for the test project to reproduce the issue.
I created a derived test that tests the same without using Spring:

```java
package test;

import org.junit.jupiter.api.Test;
import picocli.CommandLine;
import picocli.CommandLine.*;

import static org.junit.jupiter.api.Assertions.*;
import static picocli.CommandLine.ScopeType.INHERIT;

/**
 * https://github.com/remkop/picocli/issues/1772
 */
public class Issue1772 {
    public CommandLine getTestCommandLine(ParentTestCommand parentCommand, TestCommand testCommand, IFactory factory) {
        CommandLine commandLine = new CommandLine(parentCommand, factory);
        commandLine.addSubcommand(testCommand);
        return commandLine;
    }
    @Command(name = "parentTestCommand", mixinStandardHelpOptions = true, scope = INHERIT)
    static class ParentTestCommand {
    }

    @Command(name = "test")
    static class TestCommand {
        @Option(names = "-r")
        public boolean option1;

        public static String testString;

        @Command(name = "subcommand")
        void start(@Option(names = "-s") boolean option2) {
            testString = "------> -r is " + option1 + " and -s is " + option2 + " <------";
        }
    }

    @Test
    public void testIssue1772() {
        CommandLine commandLine = getTestCommandLine(new ParentTestCommand(), new TestCommand(), CommandLine.defaultFactory());
        commandLine.execute("test", "-r", "subcommand", "-s");
        assertEquals("------> -r is true and -s is true <------", TestCommand.testString);
    }
}
```

The above test fails with picocli 4.6.3, as demonstrated when I add it to the test project you provided:

```
Issue1772 > testIssue1772() FAILED
    org.opentest4j.AssertionFailedError: expected: <------> -r is true and -s is true <------> but was: <------> -r is true and -s is false <------>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1141)
        at test.Issue1772.testIssue1772(Issue1772.java:40)
```

But when I run the same test with the current version of picocli in the `main` branch (which has the fix for #1741), then the same test passes.

So, I believe this ticket can also be closed: since it won't be a problem after the fix for #1741 is released in the next version of picocli.
